sw/device/exts/common/flash_crt.S

```diff
@@ -83,6 +83,26 @@ data_copy_loop:
   blt  t0, t1, data_copy_loop
 data_copy_loop_end:
 
+  // Call the functions in the `.init_array` section.
+  //
+  // This section is typically empty except for executables built with LLVM
+  // coverage enabled. When coverage is enabled, the compiler emits pointers to
+  // the functions that initialize the profile buffer in this section. These
+  // functions must be called before the instrumented functions in the program.
+  //
+  // We use `t0` and `t1` to represent the start and end pointers of
+  // `.init_array`, respectively, and `t2` to store the addresses of the
+  // functions to be called.
+  la   t0, _init_array_start
+  la   t1, _init_array_end
+  bgeu t0, t1, init_array_loop_end
+init_array_loop:
+  lw   t2, 0(t0)
+  jalr t2
```

Comment:
`t0` and `t1` are both caller-saved. Maybe consider using `s0` and `s1` to make the spilling the C function's problem?

Reply:
Good point! I actually read the table in the risc-v spec but forgot to update this. Done.